src/main.rs

```diff
@@ -0,0 +1,200 @@
+// (De-)Serialization
+extern crate serde;
+#[macro_use]
+extern crate serde_derive;
+extern crate serde_json;
+
+// HTTP
+extern crate actix_web;
+extern crate futures;
+#[macro_use]
+extern crate hyper;
+extern crate reqwest;
+
+// Logging and CLI
+extern crate chrono;
+extern crate fern;
+#[macro_use]
+extern crate log;
+
+use std::collections::HashMap;
+
+use actix_web::{http, server, App, AsyncResponder, Error, HttpMessage, HttpRequest, HttpResponse};
+use actix_web::middleware::Logger;
+
+use futures::future::Future;
+
+header! { (XCachetToken, "X-Cachet-Token") => [String] }
+
+fn hook(req: HttpRequest) -> Box<Future<Item = HttpResponse, Error = Error>> {
+    req.clone().json()                          // <- get JsonBody future
+        .from_err()
+        .and_then(move |alert: AlertHook| {  // <- deserialized value
+            info!("{:?}", alert);
+
+            //create map for json of cachet api call
+            let mut map = HashMap::new();
+            map.insert(
+                "status",
+                match alert.status {
+                    Status::Firing => alert.alerts[0].annotations.severity,
+                    Status::Resolved => 1,
+                },
+            );
+
+            //send http put to cachet
+            let client = reqwest::Client::new();
+            let response = match client.put(&format!(
+                    "{endpoint}/api/v1/components/{component}",
+                    endpoint = match std::env::var("CACHET_BASE_URL") {
+                        Ok(val) => val,
+                        Err(_) => String::from(include_str!("cachet_endpoint.txt")),
+                    },
+                    component = alert.alerts[0].annotations.component
+                ))
+                // read "Authorization: Bearer" token into "X-Cachet_Token"
+                .header(XCachetToken(match req.headers().get("Authorization") {
+                    Some(header) => match header.to_str() {
+                        Ok(header) => {
+                            let mut header_prefix = String::from(header);
+                            header_prefix.split_off(7) //cuts of the first 7 chars: "Bearer "
```

Comment:
In general, it might be safer to remove "^Bearer " or somehow assert we're doing the right thing here, rather than doing it blindly. 

In our case, it's probably fine, so let's not worry about it.

Reply:
I don't know why it seemed to work when I tried it, according to the docs of split_off it should have paniced

Reply:
Okay the thing was that I sent the "Authentication" header and not the "Authorization" header. Having an existent but less than 7 characters long header there causes a thread of the event loop thread pool to die. I don't know whether they get respawned automatically, but even the fact that we just close the TCP conn without returning an error is bad.

Reply:
I've sent a bunch (~50) requests with a borked header and it still worked afterwards. @michaelkaye is that okay for now?

Reply:
(we do get ~140 log lines for each of those requests)

Reply:
FWIW since you're depending on hyper already you can use their header parsing stuff to do this properly: https://play.rust-lang.org/?gist=2041c466b04d07c2de9d67f95da32fee&version=stable&mode=debug

Reply:
That only gives us "Bearer <token>", just what the to_str() method from actix already gives us. Also, getting from the actix header structs to a hyper header is not worth the pain for not actually getting out what we need.